src/utils/chains.ts

```diff
@@ -8,28 +8,43 @@ import {
     parsedChainConfigExtension,
 } from '~/utils/chainConfigExtension'
 import { Chain } from '~/types'
-import { ethereumNetworks } from '~/shared/utils/constants'
+import { ethereumNetworkDisplayName, defaultEthereumNetwork } from '~/shared/utils/constants'
 import formatConfigUrl from './formatConfigUrl'
 
-function getPreferredChainName(chainName: string) {
-    if (/amoy/i.test(chainName)) {
-        return 'amoy'
+/**
+ * @param chainNameOrId Chain name from displayNames or config (if string), or chainId (if number)
+ * @returns Chain name as enumerated in @streamr/config, or default ("polygon" as of 2024) if not found
+ */
+function parseChainName(chainNameOrId: string | number): string {
```

Comment:
In master you now have a `getChainKey` utility function that, for a given string or number, gives you a proper `ChainKey` value. Doesn't use display names. You can define slugs though (like `amoy` for `polygonAmoy`) but it's optional.